src/EditorFeatures/Core/AddImports/AbstractAddImportsPasteCommandHandler.cs

```diff
@@ -115,32 +117,46 @@ public void ExecuteCommand(PasteCommandArgs args, Action nextCommandHandler, Com
                 return;
             }
 
-            using var _ = executionContext.OperationContext.AddScope(allowCancellation: true, DialogText);
-            var cancellationToken = executionContext.OperationContext.UserCancellationToken;
+            var indicatorFactory = document.Project.Solution.Workspace.Services.GetRequiredService<IBackgroundWorkIndicatorFactory>();
+            var backgroundWorkContext = indicatorFactory.Create(
```

Comment:
why not put this in a using block?

see the code in GoToDefinitionCommandHandler for a suitable pattern on how to do the sync->async jump that does all this without the need for a TAsk.Run